src/include/btree_inline.h

```diff
@@ -1788,7 +1789,8 @@ __wt_leaf_page_can_split(WT_SESSION_IMPL *session, WT_PAGE *page)
  * are 5 items on the page.
  */
 #define WT_MAX_SPLIT_COUNT 5
-    if (page->memory_footprint > (size_t)btree->maxleafpage * 2) {
+    mem_split_threshold = (size_t)WT_MIN(btree->maxleafpage * 2, btree->splitmempage);
```

Comment:
I don't think this part of the change is useful - it is targeted at cases where the leaf page size is as large as the memory page max, which makes me think that memory_page_max is small. In such cases altering the subsequent check should be enough - otherwise this path will always be taken and pages will be split with very little content.